### DISCUSSION TITLE
Generic function that takes no arguments but returns a generic type is flagged with: reportInvalidTypeVarUse

### DISCUSSION THREAD

**Discussion by stevej2608** [microsoft/pyright#7247] [Q&A]
*snippet.py*
```
def max1[T](args: list[T]) -> T:
    ...

def max2[T]() -> T:
    ...
```

In the above snippet pyright  says *max1* is OK, but *max2* is flagged with:
```
TypeVar "T" appears only once in generic function signature
Use "object" instead  (reportInvalidTypeVarUse)
```

Why is a generic functions that take no arguments unacceptable?

Python 3.12.0
pyright version 1.1.348

**Comment by erictraut** [comment]
This behavior is intended. There's no reason to use a generic type in `max2`. You should simply change the return type annotation to `object`.
